fcrepo-clients/src/test/java/edu/unc/lib/dl/fcrepo4/RepositoryObjectCacheLoaderTest.java

```diff
@@ -0,0 +1,242 @@
+    }
+
+    @Test
+    public void loadFileObjectTest() throws Exception {
+
+        mockResponseBodyWithType(pid, Cdr.FileObject);
+
+        RepositoryObject obj = objectCacheLoader.load(pid);
+
+        assertTrue(obj instanceof FileObject);
+        assertEquals(pid, obj.getPid());
+        assertEquals(ETAG, obj.getEtag());
+        assertTrue(obj.getResource().hasProperty(RDF.type, Cdr.FileObject));
+    }
+
+    @Test
+    public void loadFolderObjectTest() throws Exception {
+
+        mockResponseBodyWithType(pid, Cdr.Folder);
+
+        RepositoryObject obj = objectCacheLoader.load(pid);
+
+        assertTrue(obj instanceof FolderObject);
+        assertEquals(pid, obj.getPid());
+        assertEquals(ETAG, obj.getEtag());
+        assertTrue(obj.getResource().hasProperty(RDF.type, Cdr.Folder));
+    }
+
+    @Test
+    public void loadCollectionObjectTest() throws Exception {
+
+        mockResponseBodyWithType(pid, Cdr.Collection);
+
+        RepositoryObject obj = objectCacheLoader.load(pid);
+
+        assertTrue(obj instanceof CollectionObject);
+        assertEquals(pid, obj.getPid());
+        assertEquals(ETAG, obj.getEtag());
+        assertTrue(obj.getResource().hasProperty(RDF.type, Cdr.Collection));
+    }
+
+    @Test
+    public void loadContentRootObjectTest() throws Exception {
+
+        mockResponseBodyWithType(pid, Cdr.ContentRoot);
+
+        RepositoryObject obj = objectCacheLoader.load(pid);
+
+        assertTrue(obj instanceof ContentRootObject);
+        assertEquals(pid, obj.getPid());
+        assertEquals(ETAG, obj.getEtag());
+        assertTrue(obj.getResource().hasProperty(RDF.type, Cdr.ContentRoot));
+    }
+
+    @Test
+    public void loadAdminUnitTest() throws Exception {
+
+        mockResponseBodyWithType(pid, Cdr.AdminUnit);
+
+        RepositoryObject obj = objectCacheLoader.load(pid);
+
+        assertTrue(obj instanceof AdminUnit);
+        assertEquals(pid, obj.getPid());
+        assertEquals(ETAG, obj.getEtag());
+        assertTrue(obj.getResource().hasProperty(RDF.type, Cdr.AdminUnit));
+    }
+
+    @Test
+    public void loadBinaryObjectTest() throws Exception {
+
+        mockResponseBodyWithType(pid, Fcrepo4Repository.Binary);
+
+        RepositoryObject obj = objectCacheLoader.load(pid);
+
+        assertTrue(obj instanceof BinaryObject);
+        assertEquals(pid, obj.getPid());
+        assertEquals(ETAG, obj.getEtag());
+        assertTrue(obj.getResource().hasProperty(RDF.type, Fcrepo4Repository.Binary));
+    }
+
+    @Test(expected = ObjectTypeMismatchException.class)
+    public void loadUnsupportedTypeTest() throws Exception {
+        mockResponseBodyWithType(pid, Fcrepo4Repository.Pairtree);
+
+        objectCacheLoader.load(pid);
+    }
+
+    @Test(expected = ObjectTypeMismatchException.class)
+    public void loadInvalidPidTest() throws Exception {
+        mockResponseBodyWithType(pid, Fcrepo4Repository.Pairtree);
```

Comment:
might be nice to have a valid object type here to isolate the pid as the problem